rnd/autogpt_builder/src/components/monitor/FlowRunInfo.tsx

```diff
@@ -34,12 +40,19 @@ export const FlowRunInfo: React.FC<
             Run ID: <code>{flowRun.id}</code>
           </p>
         </div>
-        <Link
-          className={buttonVariants({ variant: "outline" })}
-          href={`/build?flowID=${flow.id}`}
-        >
-          <Pencil2Icon className="mr-2" /> Edit Agent
-        </Link>
+        <div className="flex space-x-2">
+          {flowRun.status === "running" && (
+            <Button onClick={handleStopRun} variant="destructive">
+              <IconSquare className="mr-2" /> Stop Run
+            </Button>
```

Comment:
```suggestion
            <form action={handleStopRunWithIds}>
              <Button variant="destructive">
                <IconSquare className="mr-2" /> Stop Run
              </Button>
            <form>
```

Then above in the code segment do this

```ts
const handleStopWithIds = handleStopRun.bind(null, flow.id, flowRun.id)
```

```suggestion

async function handleStopRun(flowId, runId){
    "use server";
    const api = new AutoGPTServerAPI();
    api.stopGraphExecution(flowId, runId);
  }
  ```

Reply:
why?

Reply:
So that we don’t leak our api to the frontend?